### ISSUE TITLE
Proposal: Evaluate Transforms FCurves node

### ISSUE THREAD

**Issue by jekor** [JacquesLucke/animation_nodes#1842]
In addition to the `Evaluate FCurve` node, it would be useful to have a node that takes a list of FCurves (i.e. the output of `FCurves from Object`) and a frame number as inputs and returns a transformation matrix. Any non-transform FCurves would be ignored and any missing transform FCurves would be assumed to be 0.

Use case: applying actions at different times/composing animations. As I understand it, using the `Object Action Output` node on an object more than once (i.e. at different frames) will apply only one of the actions, and that this is a fundamental limitation of the node graph evaluation algorithm and/or Blender's one-action-per-object model. Additionally, it doesn't look like actions can be composed with other animations (actions can be chained or overlayed with other actions only, and not with other animation nodes that output a time).

Using the `Evaluate FCurve` node would mean relying on the ordering of the FCurve list or having many switch nodes to route the correct FCurve to the correct transform component.

**Comment by jekor** [comment]
Here's the Script node that I'm using currently:

![1648826875](https://user-images.githubusercontent.com/136132/161294556-0e961959-7dbe-4ea7-b4ec-59d19d73b216.png)

```python
t = {
    'location': [0,0,0],
    'rotation_euler': [0,0,0],
    'scale': [0,0,0],
}

for fcurve in FCurves:
    if fcurve.data_path in t:
        t[fcurve.data_path][fcurve.array_index] = fcurve.evaluate(Frame)

Translation = Vector(tuple(t['location']))
Rotation = Euler(tuple(t['rotation_euler']))
Scale = Vector(tuple(t['scale']))
```

I don't know if 3 outputs is preferred over a single matrix for new nodes.

**Comment by OmarEmaraDev** [comment]
Sounds reasonable, a matrix output would be best I think.

**Comment by OmarEmaraDev** [comment]
Implemented in #1843.
